pkg/kapp/app/recorded_app.go

```diff
@@ -162,18 +162,26 @@ func (a *RecordedApp) CreateOrUpdate(labels map[string]string) error {
 		}
 
 		configMap.ObjectMeta.Name = a.fqName
+		// setting migration as true as we will be creating configmap with suffix
+		a.isMigrated = true
```

Comment:
Started setting `isMigrated` flag to true here as we will be creating configmap with fully qualified name and in case of `--diff-run` we unable to get the configmap and the flag will never set as we do [here](https://github.com/vmware-tanzu/carvel-kapp/blob/d667147b688102aea0f729741c2841c35e1015d2/pkg/kapp/app/recorded_app.go#L481).